src/server.ts

```diff
@@ -102,18 +102,98 @@ export class SubscriptionServer {
   private onConnect: Function;
   private onDisconnect: Function;
   private wsServer: WebSocket.Server;
-  private subscriptionManager: SubscriptionManager;
-  private executor: Executor;
+  private execute: ExecuteReactiveFunction;
   private schema: GraphQLSchema;
   private rootValue: any;
 
   public static create(options: ServerOptions, socketOptions: WebSocket.IServerOptions) {
     return new SubscriptionServer(options, socketOptions);
   }
 
-  constructor(options: ServerOptions, socketOptions: WebSocket.IServerOptions) {
-    const {subscriptionManager, executor, schema, rootValue, onSubscribe, onUnsubscribe, onRequest,
-      onRequestComplete, onConnect, onDisconnect, keepAlive} = options;
+  // TODO: All of the adapters should actually live inside static class
+  private executeFromExecute(execute: ExecuteFunction): ExecuteReactiveFunction {
+    return (schema: GraphQLSchema,
+            document: DocumentNode,
+            rootValue?: any,
+            contextValue?: any,
+            variableValues?: {[key: string]: any},
+            operationName?: string,
+      ) => ({
+        subscribe: (observer) => {
+          // TODO: if subscription call observer.error with not supported.
```

Comment:
Do u think that is a good a idea to use this.isASubscriptionRequest method here?